### PR TITLE
Fix NATReconciler to modify NATTable of NIC belonging to NATs Network

### LINKED ISSUE #181: `NATReconciler` should only modify the `NATTable` of `NetworkInterfaces` belonging to the NATs `networkRef`
**Describe the bug**
It seems like the `NATReconciler` is applying the NATs public IP to all NICs in the namespace of a given `NATGateway`. That leads to a problem if you have multiple `Networks` but only one is using a NAT. In the current `NATTable` logic the we are doing a `client.List` on all in-namespace NICs which leads to a situation where all NICs regardless of their `Network` having a NAT or not will get a public IP.

/cc @hardikdr

### PR DESCRIPTION
**Pull request by kasabe28**
# Proposed Changes

- Modify NATTable logic to list all in-namespace NICs belonging to NATs Network
- Modify NATReconciler test to create network without NAT and NIC referring to without NAT network. And validate result to have only NAT Network NIC to be in NATTable.

Fixes #181

### CODE REVIEW

File: internal/controllers/natgateway_controller.go [hunk 1]
```diff
@@ -159,13 +159,34 @@ func (r *NATGatewayReconciler) manageNATTable(
...
 	existingAllocByNicID map[types.UID]natIPAllocation,
 ) (used, requests int64, err error) {
+	natGatewayList := &v1alpha1.NATGatewayList{}
+	if err := r.List(ctx, natGatewayList,
+		client.InNamespace(natGateway.Namespace),
+	); err != nil {
+		return 0, 0, fmt.Errorf("error listing nat gatway: %w", err)
+	}
+
+	networkList := make(map[string]struct{})
```
Comment (afritzler) [inline internal/controllers/natgateway_controller.go:169]: I would suggest to use `NetworkInterface` as the value type of the `map`.

**Review by afritzler — CHANGES_REQUESTED** [review]

File: cmd/controller-manager/main.go [hunk 2]
```diff
@@ -165,6 +166,12 @@ func main() {
 		os.Exit(1)
 	}
 
+	ctx := ctrl.SetupSignalHandler()
```
Comment (afritzler) [inline cmd/controller-manager/main.go:169]: Can we move the field indexer setup right before the reconciler setup.

**Review by afritzler — CHANGES_REQUESTED** [review]
Just some cosmetics. Apart from that the PR looks good.

File: cmd/controller-manager/main.go (hunk 2 above)
Comment (kasabe28) [reply cmd/controller-manager/main.go]: Done

**Review by afritzler — APPROVED** [review]

**Pull request merged**